assignment/core/src/main/java/com/assignment/core/crypto/AESCipherProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.assignment.core.crypto;
+
+import org.osgi.service.component.annotations.Component;
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+@Component(
+        service = AsymmetricCipherProvider.class,
+        property = {
+                "algorithm=AES"
+        }
+)
+public class AESCipherProvider implements AsymmetricCipherProvider {
+    private static final String ALGORITHM = "AES";
+    private SecretKey secretKey;
+
+    public AESCipherProvider() {
+        try {
+            KeyGenerator keyGen = KeyGenerator.getInstance(ALGORITHM);
+            keyGen.init(256);
+            this.secretKey = keyGen.generateKey();
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Failed to initialize AES key", e);
```

Comment:
Is throwing an unchecked exception here going to do us any good ?

Reply:
removed unchecked expections